roles/rsyslog/tasks/outputs/elasticsearch/main.yml

```diff
@@ -25,15 +25,15 @@
         - '{{ rsyslog_output_elasticsearch }}'
       when: item.key_src | d()
 
-    - name: Check certs - triplets are not given
+    - name: Check certs - use_cert is true, but triplets are not given
       fail:
-        msg: "Error: you must specify all 3 of ca_cert, cert, key, or all 3 of ca_cert_src, cert_src, key_src in the elasticsearch output named {{ item.name }}"
+        msg: "Error: you must specify all 3 of ca_cert, cert, key, or all 3 of ca_cert_src, cert_src, key_src, or set use_cert: false in the elasticsearch output named {{ item.name }}"
       with_items:
         - '{{ rsyslog_output_elasticsearch }}'
       when:
         - item.use_cert | d(true)
-        - not ((item.ca_cert | d() and item.cert | d() and item.key | d()) or
-               (item.ca_cert_src | d() and item.cert_src | d() and item.key_src | d()))
+        - not (item.ca_cert | d() or item.cert | d() or item.key | d())
+        - not (item.ca_cert_src | d() or item.cert_src | d() or item.key_src | d())
```

Comment:
I don't think this new logic is correct.  I think the old logic is correct - fail if user does not supply all 3 of ca_cert/cert/key, or if the user does not supply all of ca_cert_src/cert_src/key_src

Reply:
I think this original code [1] works without ca_cert, cert and key. They local files ca_cert_src, cert_src and key_src are copied to {{ rsyslog_config_dir }} on the target host. And the paths are set to the rsyslog config params tls.cacert, tls.mycert, and tls.myprivkey, respectively, (I noticed there was an issue in the original code, e.g., if the file names are not es-ca.crt, es-cert.pem, or es-key.pem, rsyslog or if {{ rsyslog_config_dir }} is different from /etc/rsyslog.d, rsyslogd should issue the certs not found error, but it's not a point here :) The point is setting ca_cert_src, cert_src and key_src without ca_cert, cert and key was supported prior to this PR. Do you think it's not useful?
[1] - https://github.com/linux-system-roles/logging/blob/master/roles/rsyslog/tasks/outputs/elasticsearch/main.yml#L2-L53

Reply:
> The point is setting ca_cert_src, cert_src and key_src without ca_cert, cert and key was supported prior to this PR. Do you think it's not useful?

It is useful.  But I don't think that is related to the problem I see with the current code.  This is the current code:
```yaml
     when:
        - not (item.ca_cert | d() or item.cert | d() or item.key | d())
        - not (item.ca_cert_src | d() or item.cert_src | d() or item.key_src | d())
```
iiuc, this means that the user can set `item.ca_cert`, but not set `item.cert` or `item.key`, and the user can set `item.cert_src`, but not set `item.ca_cert_src` or `item.key_src`, and the `when` will report `true`.  Is this intended?

Reply:
Thanks, @richm. My mistake. The fix was pushed.